### ISSUE TITLE
ERROR: runPCA.r

### ISSUE THREAD

**Issue by DamienTan** [rounakdey/FastSparseGRM#1]
Dear rounakdey,
When I used the script `[runPCA_wrapper.R](https://github.com/rounakdey/FastSparseGRM/blob/main/extdata/runPCA_wrapper.R)` to run PCA, there were not output results. And I checked the `runPCA.Rout`:
***
> library('FastSparseGRM')
> runPCA(prefix.in,file.unrels,no_pcs,num_threads,prefix.out,no_iter)
[1] "PCA started at 2022-12-04 17:53:17"
[1] "4 relateds, 768 unrelateds, and 24928532 variants"
Reading the file: pruned.bed
Reading BED file complete 
Number of SNPs read: 24928532
Number of samples: 772
Using 40 threads
Processing samples complete 
Allele frequencies calculated 
[1] "Reading Completed in 16 seconds"
[1] "1 iterations completed"
[1] "2 iterations completed"
[1] "3 iterations completed"
[1] "4 iterations completed"
[1] "5 iterations completed"
[1] "6 iterations completed"
[1] "7 iterations completed"
[1] "8 iterations completed"
[1] "9 iterations completed"
[1] "10 iterations completed"
[1] "Iterations Completed in 683 seconds"
Error in svd(H, nv = 0) : infinite or missing values in 'x'
Calls: runPCA -> drpca -> svd
Execution halted
***
It seems there are infinite or missing values, what I should do to solve this problem? 
Hoping to receive your reply!

**Comment by rounakdey** [comment]
Hi @DamienTan, I think this error is because of the presence of monomorphic SNPs in the data. FastSparseGRM PCA method does not currently handle monomorphic SNPs. Please filter the input BED file so that the MAC is at least one. Hopefully this should resolve the problem. Best, Rounak

**Comment by DamienTan** [comment]
> Hi @DamienTan, I think this error is because of the presence of monomorphic SNPs in the data. FastSparseGRM PCA method does not currently handle monomorphic SNPs. Please filter the input BED file so that the MAC is at least one. Hopefully this should resolve the problem. Best, Rounak

Thanks Rounak, 
I deleted all the missing genotypes which are marked as "./." in vcf file, and run PCA again. 
It worked.

**Comment by rounakdey** [comment]
Good to know. I think you don't need to delete all missing genotypes, just the ones that are causing the SNPs to have MAC = 0 in the dataset. But anyway, as the problem is solved, I will mark this issue as closed. Thanks.

**Comment by daniel-hui** [comment]
Hi Rounak,

Thanks for making this package. I also had this issue but it was resolved based on suggestions in this thread. Additionally I had to remove MAC=0 SNPs when subsetting to the unrelateds only -- it worked without removing missing genotypes.

This is sort of unrelated and not that you asked but it may be helpful to have some installation instructions on the main README, I was able to install it like:
```
library(devtools)
install_github("rounakdey/FastSparseGRM")
```

**Comment by DamienTan** [comment]
Just for supplement. After I re-ran this `runPCA` step using more samples, this error occurred again.
```
Error in svd(H, nv = 0) : infinite or missing values in 'x'
Calls: runPCA -> drpca -> svd
```
The filtered.bed file (MAF < 0.05 AND pruning) is used, maybe no need 'MAC=0' cause I have already run MAF<0.05. I removed all the missing genotype last time and it actually worked, but it seems a bit extreme. 
I noticed some SNPs with a high missing rate in my data. I tried to remove these low quality SNPs using plink `--geno 0.01` first, and other filter options(MAF and pruning). 
Glad that it worked.
Maybe the problem is so many missing genotypes in one SNP, and we don't need to remove all the SNPs which contain one or more missing genotypes(./.). Wish there are other good explanations or solutions and hope what I submitted can help other people who have a similar problem when they use this tool.
